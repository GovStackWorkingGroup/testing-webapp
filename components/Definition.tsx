import React, { useCallback } from 'react';
import { useIntl } from 'react-intl';
import ReactMarkdown from 'react-markdown';

const Definition = () => {
  const { formatMessage } = useIntl();
  const format = useCallback(
    (id: string) => formatMessage({ id }),
    [formatMessage]
  );

  return (
    <div className='definition-section'>
      <p className='definition-title' data-testid='definition-title'>
        {format('app.definition.title')}
      </p>
      <div data-testid='definition-description'>
        <ReactMarkdown className='definition-description' linkTarget="_blank">
          {format('app.definition.description')}
        </ReactMarkdown>
      </div>
      <p className='definition-note' data-testid='definition-note'>
        {format('app.definition.note')}
      </p>
    </div>
  );
};

export default Definition;
