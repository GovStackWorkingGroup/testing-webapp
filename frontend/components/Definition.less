@import '../styles/Variables.less';

.definition-section {
  width: 70vw;
  display: flex;
  flex-direction: column;
  gap: 10px;
  padding-top: 40px;
  padding-bottom: 40px;
}

.definition-section-software-compliance {
  width: 90vw;
}

.definition-section p {
  text-align: justify;
}

.definition-title {
  font-size: 18px;
  color: @text-nav-and-btn;
  font-weight: 700;
}

.definition-description a {
  color: @accents-and-btn;
  text-decoration: none;
}

.definition-description a:hover {
  text-decoration: underline;
}

.definition-note {
  font-style: italic;
}
