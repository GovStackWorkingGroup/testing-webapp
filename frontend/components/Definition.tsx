import React from 'react';
import ReactMarkdown from 'react-markdown';
import classNames from 'classnames';
import { formatTranslationType } from '../service/types';
import children = ReactMarkdown.propTypes.children;

type DefinitionType = {
  title: formatTranslationType;
  hasRedirecting?: boolean;
  description?: formatTranslationType;
  descriptionPartTwo?: formatTranslationType;
  note?: formatTranslationType;
  customStyle?: string;
  children?: React.ReactNode;
};

const Definition = ({
  title,
  hasRedirecting = false,
  description,
  descriptionPartTwo,
  note,
  customStyle,
  children
}: DefinitionType) => (
  <div className={classNames('definition-section', customStyle)}>
    <p className="definition-title" data-testid="definition-title">
      {title}
    </p>
    <div data-testid="definition-description">
      {hasRedirecting ? (
        <div>
          <ReactMarkdown className="definition-description" linkTarget="_blank">
            {`${description}`}
          </ReactMarkdown>
          <ReactMarkdown className="definition-description" linkTarget="_blank">
            {`${descriptionPartTwo}`}
          </ReactMarkdown>
        </div>
      ) : (
        <div className="definition-description">{description}</div>
      )}
      {children}
    </div>
    {note && (
      <p className="definition-note" data-testid="definition-note">
        {note}
      </p>
    )}
  </div>
);

export default Definition;
