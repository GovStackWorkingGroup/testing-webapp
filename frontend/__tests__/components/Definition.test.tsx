import Definition from '../../components/Definition';
import { render } from '../test-utils/test-utils';

describe('Unit tests for Definition component:', () => {
  const TITLE_TEST_ID = 'definition-title';
  const DESCRIPTION_TEST_ID = 'definition-description';
  const NOTE_TEST_ID = 'definition-note';

  const mockedTitle = 'Test title';
  const mockedDescription = 'Test description';
  const mockedNote = 'Test note';

  it('should render the component and match snapshot', () => {
    const { container } = render(
      <Definition title={mockedTitle} description={mockedDescription} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should have corresponding values', () => {
    const { getByTestId } = render(
      <Definition
        title={mockedTitle}
        description={mockedDescription}
        note={mockedNote}
      />
    );

    expect(getByTestId(TITLE_TEST_ID)).toHaveTextContent(mockedTitle);
    expect(getByTestId(DESCRIPTION_TEST_ID)).toHaveTextContent(
      mockedDescription
    );
    expect(getByTestId(NOTE_TEST_ID)).toHaveTextContent(mockedNote);
  });
});
