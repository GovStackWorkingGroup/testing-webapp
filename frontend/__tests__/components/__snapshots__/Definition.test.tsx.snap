// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Definition component: should render the component and match snapshot 1`] = `
<div>
  <div
    class="definition-section"
  >
    <p
      class="definition-title"
      data-testid="definition-title"
    >
      Test title
    </p>
    <div
      data-testid="definition-description"
    >
      <div
        class="definition-description"
      >
        Test description
      </div>
    </div>
  </div>
</div>
`;
